### PR TITLE
Introduce testbed examples starting with 2d

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Make progress for #15918 
- Start with 2d

## Solution

- Remove screenshots for existing examples as they're not deterministic
- Create new "testbed" example category, with a 2d one to start

## Testing

- Run `CI_TESTING_CONFIG=.github/example-run/testbed_2d.ron cargo run --example testbed_2d --features "bevy_ci_testing"`
- ???
- Check the screenshots

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
I'm so pleased to have this started. An easy to use single testbed example (or three) will make testing changes so much easier. We should probably migrate our existing `ui` example over to this at some point.

File: examples/testbed/2d.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    keyboard: Res<ButtonInput<KeyCode>>,
+    #[cfg(feature = "bevy_ci_testing")] mut ci_events: ResMut<Events<CiTestingCustomEvent>>,
+    scene: Res<State<Scene>>,
+    mut next_scene: ResMut<NextState<Scene>>,
+) {
+    let mut should_switch = false;
+    should_switch |= keyboard.just_pressed(KeyCode::Space);
+    #[cfg(feature = "bevy_ci_testing")]
+    {
+        should_switch |= !ci_events.is_empty();
```
Comment (BD103) [inline examples/testbed/2d.rs:41]: I would prefer this actually filter for the `switch_scene` command, so it doesn't break when other custom commands are added in the future.

File: examples/testbed/2d.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    }
+    if should_switch {
+        info!("Switching scene");
+        #[cfg(feature = "bevy_ci_testing")]
+        ci_events.clear();
```
Comment (BD103) [inline examples/testbed/2d.rs:46]: If you switch to filtering, this should be removed.

**Review by BD103 — APPROVED** [review]
Looks good! I'd prefer to actually filter for the `switch_scene` command, but won't block on it.

**Pull request merged**